unittests/misc_tests.cpp

```diff
@@ -826,6 +826,12 @@ BOOST_AUTO_TEST_CASE(transaction_test) { try {
    BOOST_CHECK( packed != fc::raw::pack( static_cast<const transaction&>(pkt8.get_transaction()) ));
    BOOST_CHECK( packed == pkt8.get_packed_transaction() ); // extra maintained
 
+   // Check extra maintained in transitions of v0 -> v1 -> v0
+   packed_transaction pkt8_v0_to_v1(pkt8, true); // Build v1 with extra
+   auto pkt8_v0_to_v1_to_v0 = pkt8_v0_to_v1.to_packed_transaction_v0();
+   BOOST_CHECK_EQUAL(pkt.get_transaction().id(), pkt8_v0_to_v1_to_v0->get_transaction().id());
+   BOOST_CHECK( packed != fc::raw::pack( static_cast<const transaction&>(pkt8_v0_to_v1_to_v0->get_transaction()) ));
+   BOOST_CHECK( packed == pkt8_v0_to_v1_to_v0->get_packed_transaction() ); // extra maintained
```

Comment:
- Add tests for context_free_data being maintained.
- Add tests for get_prunable_size and get_unprunable_size being equal.
- Add tests for empty context free data going through conversion. This tests with context_free_data but not with an empty context free data. We saw indication that it could be handled differently. Would be good to repeat all the above tests with empty context free data.
- Add test where you start with a packed_transaction with extra data and make the round trip.

Reply:
Done.